hw/hw13/trigrams.py

```diff
@@ -0,0 +1,87 @@
+import string
+import random
+
+
+def strip_cover(infile):
+    #  strip header and footer, then store the whole book as a string
+    wholebook = infile.read(-1)
+    start_string = 'SIR ARTHUR CONAN DOYLE'
+    end_string = 'End of the Project Gutenberg EBook'
+    start = wholebook.find(start_string)
+    end = wholebook.find(end_string)
+    content = wholebook[start + len(start_string):end]
+    return content
+
+
+def break_paragraph_into_list(infile):
+    # break the whole book into list of paragraphs
+    content = strip_cover(infile)
+    # DOS text
+    # plist = re.split('\n\n', content)
+    plist = content.split('\n\n')
+    return plist
+
+
+def scrub_text(infile):
+    # clean up text until only words are left in each paragraph
+    plist = break_paragraph_into_list(infile)
+    delchars = string.punctuation
+    # remove any paragraph that is less than 3 words, or empty
+    for e in plist:
+        if (len(e.strip().split()) < 3):
+            plist.remove(e)
+    for i in range(len(plist)):
+        p = plist[i]
+        # remove white spaces and line break
+        p = p.strip().replace('\n', ' ')
+        # replace -- with space, and preserve hyphenation
+        p = p.replace('--', ' ').replace('-', '')
+        # remove puctuation
+        p = p.translate(str.maketrans('', '', delchars))
+        # store lowercase scrubbed text back into list
+        plist[i] = p.lower()
+    return plist
+
+
+def store_trigram(infile):
+    # build a dictionary of trigrams
+    plist = scrub_text(infile)
+    t = {}
+    w = []
```

Comment:
Try to avoid using single character variables. Again, [refer to the google python style guide](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Naming#Naming). Once python programs become large, it can be very difficult to debug errors that contain a bunch of 1 letter variables when you're not sure what they represent. Consider when other people may be needing to work on your program in the future.